auth/authenticate-middleware.js

```diff
@@ -2,7 +2,24 @@
   complete the middleware code to check if the user is logged in
   before granting access to the next middleware/route handler
 */
+const jwt = require('jsonwebtoken');
 
-module.exports = (req, res, next) => {
-  res.status(401).json({ you: 'shall not pass!' });
-};
+
+const authenticated = (req, res, next) => {
+  const token = req.headers.authentication;
+
+  if (token) {
+    jwt.verify(token, process.env.JWT_SECRET, (err, decodedToken) => {
+      if (err) {
+        res.status(401).end()
+      } else {
+        req.decodeJwt = decodedToken;
+        next();
+      }
+    })
+  } else {
+    res.status(403).end()
```

Comment:
I think 401 is more appropriate here.  As not having 401 means not authorized, and 403 means I know who you are but you don't have permission